presto-main/src/main/java/com/facebook/presto/sql/planner/SubqueryPlanner.java

```diff
@@ -305,6 +305,7 @@ private PlanBuilder appendExistSubqueryApplyNode(PlanBuilder subPlan, ExistsPred
 
         VariableReferenceExpression exists = variableAllocator.newVariable(getSourceLocation(existsPredicate), "exists", BOOLEAN);
         subPlan.getTranslations().put(existsPredicate, exists);
+        // TODO: Figure out a better way to write this predicate, since Exists(true) isn't a valid expression
```

Comment:
exists select 1?

Reply:
That will not work as optimizer doesn't expect subquery(select 1) as expressions. These are meant to be removed by the SubqueryPlanner. It will be cleaner once we manage to directly write a RowExpression here instead of Expression. I updated the comment